test/integration/lib/make_suite.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+const Config = require('config-js');
+
+var config = new Config('./test/integration/lib/config/foxbox.js');
+var philipshue_server = require('../lib/philipsHue_server.js');
+var ipcamera_server = require('../lib/ipcamera_server.js');
+var webPush_server = require('../lib/webpush_server.js');
+var nupnp_server = require('../lib/nupnp_PhilipsHue.js');
+var foxboxManager = require('../lib/foxbox_process_manager.js');
+var config = new Config('./test/integration/lib/config/foxbox.js');
+
+
+var make_suite = (function() {
+  var nupnpSimulatorOn = false;
+  var hueSimulatorOn = false;
+  var cameraSimulatorOn = false;
+  var webPushSimulatorOn = false;
+  
+  function turnOnAllSimulators() { 
+
+    // Note: when one of the simulators needs to be enabled later, make sure
+    // it is in the before hook with the turnOnxxx method returning the promise
+    var promises = [];
+
+    promises.push(turnOnHueNupnp());
+    promises.push(turnOnHue());
+    promises.push(turnOnCamera());
+    promises.push(turnOnWebPush());
+    
+    return Promise.all(promises);
+ }
+
+  function turnOnHueNupnp() {
+      nupnpSimulatorOn = true;
+      return nupnp_server.start(config.get('nupnp_server.id'),
+      config.get('philips_hue.url') +':'+config.get('philips_hue.port'),
+      config.get('nupnp_server.port'));
+  }
+
+  function turnOnHue(){
+    hueSimulatorOn = true;
+  
+    return philipshue_server.setup(config.get('philips_hue.port'));
+    
+  }
+
+  function turnOnCamera() {
+    cameraSimulatorOn = true;
+    return ipcamera_server.setup();
+  }
+
+  function turnOnWebPush() {
+    webPushSimulatorOn = true;
+    return webPush_server.setup();
+  }
+
+  function turnOnFoxbox(done) {
+    foxboxManager.fullOptionStart(done);
```

Comment:
thought about it, but I since it doesn't have any async, i thought it'd be a bit misleading, plus I couldn't find a easy way to add promise that resolves on callbacks.